baseapp/baseapp.go

```diff
@@ -876,20 +881,22 @@ func (app *BaseApp) prepareProposal(req abci.RequestPrepareProposal) ([][]byte,
 	return txsBytes, nil
 }
 
-func (app *BaseApp) processProposal(req abci.RequestProcessProposal) error {
-	for _, txBytes := range req.Txs {
-		tx, err := app.txDecoder(txBytes)
-		if err != nil {
-			return err
-		}
+func (app *BaseApp) DefaultProcessProposal() sdk.ProcessProposalHandler {
+	return func(ctx sdk.Context, req abci.RequestProcessProposal) abci.ResponseProcessProposal {
+		for _, txBytes := range req.Txs {
+			tx, err := app.txDecoder(txBytes)
+			if err != nil {
+				panic(err)
```

Comment:
This doesn't seem right to me.
If a byzantine proposer sends you malformed TXs that you can't decode, you shouldn't crash. That's DoS.
Shouldn't you return "Reject" here?

Reply:
Oops! How embarrassing, I broke this in a refactor.